### PR TITLE
Change ManagedCluster's short names

### LINKED ISSUE #364: Fix ManagedCluster CRD confusing points
I noticed several issues with the `ManagedCluster` CR:

1. Name `ManagedCluster` clashes with the resource `ManagedCluster` of  `containerservice.azure.com/v1api20210501` which is a part of Azure provider. This means that user can't execute `kubectl get managedcluster` and has to user the full name `kubectl get managedcluster.v1alpha1.hmc.mirantis.com`
2. Short names of the `ManagedCluster` are `hmc-deploy` and `deploy` which make confusion even more.

**The task**

Replace short names to be similar to the full name. In the docs if `ManagedCluster` resource is mentioned in scope of kubectl commands it should referenced by a full name with group and version (`managedcluster.v1alpha1.hmc.mirantis.com`).

### PR DESCRIPTION
**Pull request by a13x5**
Resolves #364 
Small fix to get rid of `Deployment` resource legacy.

### CODE REVIEW

File: api/v1alpha1/managedcluster_types.go
Comment (zerospiel) [inline api/v1alpha1/managedcluster_types.go:117]: i have already done this in the #422 but the names are different (`hmcmc;mc`), if the latter are ok maybe we could wait for the merge, otherwise LGTM

**Review by zerospiel — APPROVED** [review]

File: api/v1alpha1/managedcluster_types.go
Comment (a13x5) [reply api/v1alpha1/managedcluster_types.go:117]: @zerospiel Tbh I don't have any preferences on the naming.
But I guess I will merge this to provide correct changes to the docs which I'm doing now.
Thanks!

**Pull request merged**
